tests/test_components.py

```diff
@@ -481,37 +541,80 @@ def test_parse(self):
+        self.component = Collections(self.bootstrap)
+
+    def test_parse(self):
+        """
+        Test that parsing is able to call the Collection.get_or_create method
+        """
+        component = self.component
+        component.raw_data = {'collection1': {'pages': ['page1'], 'name': 'Collection 1', 'user': 'user1', 'workflow': 'wf1'}}            
+
+        with patch.object(
+            component, 'get_version', return_value=self.version
+        ) as get_version, patch.object(
+            component, 'get_or_create'
+        ) as get_or_create, patch.object(
+            component, 'add_version'
+        ) as add_version:
+            component.parse()
+            get_or_create.assert_called_once_with({'workflow': self.wf1, 'name': 'Collection 1', 'author': self.user})
+            
+    def test_get_version(self):
+        version = self.component.get_version(self.page1)
+        self.assertEqual(self.version, version)
+
+    def test_get_or_create(self):
+        # test add
+        collection_data = {
+            "author": self.user,
+            "workflow": self.wf1,
+            "name": 'Collection 1',
+        }
+        collection = self.component.get_or_create(collection_data)
+        self.assertEqual(collection.id, 1)
+        collection_data2 = {
+            "author": self.user,
+            "workflow": self.wf1,
+            "name": 'Collection 2',
+        }
+        collection2 = self.component.get_or_create(collection_data2)
+        self.assertEqual(collection2.id, 2)
+
+        # test get
+        collection3 = self.component.get_or_create(collection_data)
+        self.assertEqual(collection.id, 1)
+
+    def test_add_version(self):
+        """
+        Test that adding a version to a collection creates 
+        a moderation request which references that version
+        """
+        collection_data = {
+            "author": self.user,
+            "workflow": self.wf1,
+            "name": 'Collection 1',
+        }
+        collection = self.component.get_or_create(collection_data)
+        version = self.component.add_version(collection, self.version)
+        moderation_request = collection.moderation_requests.first()
+        stored_version = moderation_request.version
+        # check the stored version
+        self.assertEqual(self.version, stored_version) 
+        # check the stored content
+        self.assertEqual(moderation_request.version.content, self.version.content)
+        
+    def test_integration(self):
+        component = self.component
+        component.raw_data = {'collection1': {'pages': ['page1'], 'name': 'Collection 1', 'user': 'user1', 'workflow': 'wf1'}}            
         component.parse()
-        self.assertEqual(ModerationCollection.objects.count(), 6)
+        self.assertEqual(ModerationCollection.objects.count(), 1)
+        collection = component.data["collection1"]
+        self.assertEqual(collection.name, "Collection 1")
```

Comment:
`ModerationCollection` has quite a few more fields, see here: https://github.com/FidelityInternational/djangocms-moderation/blob/release/1.0.x/djangocms_moderation/models.py#L225
The other fields could be filled in incorrectly and this test (or any other test for that matter) would not fail.

Reply:
I'm very confused by this. `test_components.py` only has 606 lines. Are you looking at an earlier version?

Reply:
You're only asserting one field (`name`) on the ModerationCollection instance (the line I commented on). You're setting more than one field on Moderation in `def parse` though.